File: src/com/l2kt/gameserver/network/clientpackets/RequestPledgeMemberInfo.java
package com.l2kt.gameserver.network.clientpackets;

import com.l2kt.gameserver.model.actor.instance.Player;
import com.l2kt.gameserver.model.pledge.Clan;
import com.l2kt.gameserver.model.pledge.ClanMember;
import com.l2kt.gameserver.network.serverpackets.PledgeReceiveMemberInfo;

/**
 * Format: (ch) dS
 * @author -Wooden-
 */
public final class RequestPledgeMemberInfo extends L2GameClientPacket
{
	@SuppressWarnings("unused")
	private int _pledgeType;
	private String _player;
	
	@Override
	protected void readImpl()
	{
		_pledgeType = readD();
		_player = readS();
	}
	
	@Override
	protected void runImpl()
	{
		final Player activeChar = getClient().getActiveChar();
		if (activeChar == null)
			return;
		
		// do we need powers to do that??
		final Clan clan = activeChar.getClan();
		if (clan == null)
			return;
		
		final ClanMember member = clan.getClanMember(_player);
		if (member == null)
			return;
		
		activeChar.sendPacket(new PledgeReceiveMemberInfo(member));
	}
}

File: src/com/l2kt/gameserver/network/clientpackets/RequestPledgeMemberInfo.kt
package com.l2kt.gameserver.network.clientpackets

import com.l2kt.gameserver.network.serverpackets.PledgeReceiveMemberInfo

/**
 * Format: (ch) dS
 * @author -Wooden-
 */
class RequestPledgeMemberInfo : L2GameClientPacket() {
    private var _pledgeType: Int = 0
    private var _player: String? = null

    override fun readImpl() {
        _pledgeType = readD()
        _player = readS()
    }

    override fun runImpl() {
        val activeChar = client.activeChar ?: return

        val clan = activeChar.clan ?: return

        val member = clan.getClanMember(_player) ?: return

        activeChar.sendPacket(PledgeReceiveMemberInfo(member))
    }
}